app/controlplane/pkg/biz/workflow.go

```diff
@@ -190,9 +209,48 @@ func (uc *WorkflowUseCase) Update(ctx context.Context, orgID, workflowID string,
 		return nil, NewErrNotFound("workflow")
 	}
 
+	// Dispatch events to the audit log regarding the contract if it has changed
+	// or if it has been attached or detached
+	if opts.ContractID != nil {
+		if preUpdateWorkflow.ContractID != uuid.Nil {
+			uc.dispatchContractEvent(ctx, orgID, preUpdateWorkflow.ContractID, preUpdateWorkflow.ContractName, wf.ID, wf.Name, false)
```

Comment:
ok, so this is detaching the previous contract and attaching it to a new one?

Reply:
yes, in case of a change of contract it sends the detach and attach. See for example when we create a new workflow, we send the event of a new contract attached